tools/autograd/gen_autograd.py

```diff
@@ -71,14 +72,22 @@
 for key in VIEW_FUNCTIONS_WITH_METADATA_CHANGE:
     VIEW_FUNCTIONS[key] = 'self'
 
+# Functions for which we use CreationMeta::MULTI_OUTPUT_SAFE. I.e., the ones for
+# which inplace modification of outputs is being gradually deprecated.
+MULTI_OUTPUT_SAFE_FUNCTIONS = {
```

Comment:
I don't think split and chunk need to be here right? They are just implemented with `split_with_sizes` and don't have any custom logic in them?

Reply:
I think that `chunk` indeed doesn't need to be here but `split` and `split_with_sizes` have separate implementations (that both call `narrow`), so I think those should remain.

Reply:
Ho very good point for `split`. But since we define a special derivative for it, we should move it from `RETURNS_VIEWS_OF_INPUT` to `VIEW_FUNCTIONS` in `tools/autograd/gen_autograd.py`.
This won't change the codegen now that you updated the generation to check both lists. But just to be consistent with the comments there.

Reply:
EDIT: You actually already added it in `VIEW_FUNCTIONS`, so it is in both right now haha